### PR TITLE
feat: nested ServeMux support for net/http

### PR DESCRIPTION
**Pull request by antst**
## Summary
- Detect hierarchical `http.ServeMux` nesting via `Handle("/prefix/", childMux)`
- Support `http.StripPrefix` wrapping transparently
- 3+ levels of nesting produce correct accumulated path prefixes
- Variable-based mux creation and function-based route registration both traced
- Package-level `http.HandleFunc` (DefaultServeMux) detected

## Changes
- `internal/spec/config.go`: Add `MountPattern` to `DefaultHTTPConfig()`
- `internal/spec/extractor.go`: Add `handleVariableMount()` — finds mux creation node by `CalleeRecvVarName`, traverses its children with mount path. StripPrefix children inspected for the actual mux variable.
- `internal/engine/engine_e2e_test.go`: Add `nested_http` to golden test matrix
- `testdata/nested_http/`: New fixture — 3-level nesting, StripPrefix, function-based registration, DefaultServeMux, edge case (same path at different levels)
- `internal/spec/extractor_mount_test.go`: 3800+ lines of tests covering `handleRouterAssignment`, `extractRouteChildren`, `handleMountNode`, `handleVariableMount`, `callArgToString`, and more
- `README.md`: net/http Mounting/Groups updated from "—" to "Nested `ServeMux`"

## Generated paths from fixture
```
/api/health       — apiMux level
/api/v1/health    — v1Mux level (edge case: same /health at two levels)
/api/v1/items     — v1Mux level (registered via helper function)
/api/v1/users     — v1Mux level (registered via helper function)
/global           — DefaultServeMux (package-level http.HandleFunc)
/ping             — rootMux level
```

## Test plan
- [x] All 14 golden files pass (7 frameworks × 2 naming modes)
- [x] 6 existing framework golden files unchanged (no regressions)
- [x] Full test suite passes (10 packages)
- [x] Lint clean (0 issues)
- [x] Coverage: 86.3% (all packages ≥ 85%)
- [x] 4 dedicated unit tests for `handleVariableMount`
- [x] Comprehensive extractor coverage tests (3800+ lines)

## Summary by CodeRabbit

* **New Features**
  * Detect and document nested net/http ServeMux mounts in generated OpenAPI output.

* **Tests**
  * Added end-to-end and unit tests plus golden fixtures to exercise nested ServeMux scenarios.

* **Documentation**
  * Updated framework compatibility table to show net/http mounting/groups support.

### CODE REVIEW

**Comment by antst** [comment]
@coderabbit review

File: testdata/nested_http/expected_openapi.json [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+          }
+        },
+        "required": [
+          "id",
+          "name"
+        ]
+      }
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (antst) [reply testdata/nested_http/expected_openapi.json:193]: @coderabbitai This is expected and not a bug. Both golden files (`expected_openapi.json` and `expected_openapi_legacy.json`) are intentionally present — the test matrix always compares both short-name and legacy-name output for every framework fixture.

They are identical here because the `nested_http` module has a flat module path (`module nested_http`), so there's no deep package path to shorten. The short-name logic (stripping `github.com/org/project/internal/...` prefixes) has nothing to strip, producing identical output.

For comparison, the `chi` or `response_patterns` fixtures have divergent golden files because their module paths are deeper. This is working as designed — the files will diverge naturally if the fixture's module path becomes more complex.
